python/cudf/cudf/core/column/column.py

```diff
@@ -1589,6 +1596,18 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
         data = as_column(
             np.asarray(arbitrary), dtype=dtype, nan_as_null=nan_as_null
         )
+    elif isinstance(arbitrary, pd.core.arrays.masked.BaseMaskedArray):
+        cudf_dtype = arbitrary._data.dtype
+
+        data = Buffer(arbitrary._data.view("|u1"))
+        data = as_column(data, dtype=cudf_dtype)
+
+        mask = arbitrary._mask
+        mask = bools_to_mask(
+            as_column(mask).binary_operator("eq", np.bool_(False))
+        )
+
+        data = data.set_mask(mask)
```

Comment:
Should move this and the other `PandasArray` branches to above the general `__array_interface__` handling to below the general `pandas.Series` handling.